modules/genomic_browser/ajax/genomic_file_upload.php

```diff
@@ -120,15 +167,28 @@ function moveFileToFS(&$fileToUpload)
 
     $config           = NDB_Config::singleton();
     $genomic_data_dir = $config->getSetting('GenomicDataPath');
-    $DB =& Database::singleton();
-    reportProgress(40, "Copying file to $genomic_data_dir ");
-    if (move_uploaded_file(
+    $validPreconditions = true;
+
+    // update ui to show we are trying to move the file
+    reportProgress(98, "Copying file to $genomic_data_dir ");
+    // file system validation
+    if (!file_exists($fileToUpload->full_path)) {
+        error_log("Specified path $fileToUpload->full_path does not exist.");
+        $validPreconditions = false;
+    } else if (!is_dir($fileToUpload->full_path)) {
+        error_log("$fileToUpload->full_path exists but is not a directory.");
+        $validPreconditions = false;
+    } else if (!is_writable($fileToUpload->full_path)) {
+        error_log("$fileToUpload->full_path is not writable by web user.");
+        $validPreconditions = false;
+    }
+    // print failure message if preconditions not met or if
+    // file move fails for any other reason
+    if (!($validPreconditions || move_uploaded_file(
```

Comment:
It's bad practice/habit to intentionally depend upon hidden side effects in an if statement evaluation.

I would move the `move_uploaded_file()` call outside of the if statement and save the return value to a variable for clarity. (Doing so might require reworking this logic a little, but it would also make it cleaner and easier to follow.)